### ISSUE TITLE
[UX] Automatically add ALT text when not provided by user.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3173]
[initial comment moved to #3006]

Interesting modules:

- https://www.drupal.org/project/auto_alter

    > The module uses the [Microsoft Azure Cognitive Services API](https://www.microsoft.com/cognitive-services) to generate an Alternative Text for images when no Alternative Text has been provided by user.

    ...real world example, no demo content:

    ![image](https://user-images.githubusercontent.com/2423362/41751031-8009961e-7602-11e8-9bd0-1886b4a92b98.png)

    The free version of the API currently allows for 5000 free transactions per month.

- https://www.drupal.org/project/google_vision

    > ...
    > 
    > 3. **Filling Alt Text attribute of image file**- Google Vision API module allows auto filling of the Alt Text attribute of the image files using Label Detection/Logo Detection/Landmark Detection/Optical Character Detection, as is chosen by the end user.

    The [respective service from Google](https://cloud.google.com/vision) allows 1000/month for free.

**Comment by BWPanda** [comment]
If this requires a third-party system to work, especially one that's limited to a certain number of uses per month, then I think this is better in contrib than core...

**Comment by klonos** [comment]
Yep, I filed this ticket before I was made aware of the existence of #3006. I have since moved all the core-worthy stuff from thins ticket here over to that other one. ...also, hence the `status - contrib candidate` status on this ticket here 😉

But fair enough all in all, this should perhaps be moved into https://github.com/backdrop-contrib as a proper request to (back)port these 2 D8 modules.

Closing...
